Sources/TuistCore/MetadataProviders/XCFrameworkMetadataProvider.swift

```diff
@@ -89,11 +89,25 @@ public final class XCFrameworkMetadataProvider: PrecompiledMetadataProvider, XCF
 
     public func binaryPath(xcframeworkPath: AbsolutePath, libraries: [XCFrameworkInfoPlist.Library]) throws -> AbsolutePath {
         let archs: [BinaryArchitecture] = [.arm64, .x8664]
-        guard let library = libraries.first(where: { !$0.architectures.filter(archs.contains).isEmpty }) else {
+        let binaryName = xcframeworkPath.basenameWithoutExt
+
+        guard let library = libraries.first(where: {
+            let validArchitectures = !$0.architectures.filter(archs.contains).isEmpty
+            guard validArchitectures, let binaryPath = try? path(for: $0, binaryName: binaryName, xcframeworkPath: xcframeworkPath) else { return false }
+            guard localFileSystem.exists(binaryPath) else {
+                let relativePath  = binaryPath.components.suffix(3).joined(separator: "/")
+                logger.warning("\(xcframeworkPath.basename) is missing architecture \(relativePath) defined in the Info.plist")
```

Comment:
I think it'd be nice to test this in the unit test via `XCTAssertPrinterOutputContains`

Reply:
Good shout, I hadn't seen those helpers. Added in the latest push.